ui/src/app/applications/components/application-pod-view/pod-view.tsx

```diff
@@ -0,0 +1,390 @@
+    public static contextTypes = {
+        apis: PropTypes.object
+    };
+    private menuItemsFor(modes: PodGroupType[], prefs: ViewPreferences): MenuItem[] {
+        const podPrefs = prefs.appDetails.podView || ({} as PodViewPreferences);
+        return modes.map(mode => ({
+            title: (
+                <React.Fragment>
+                    {podPrefs.sortMode === mode && <i className='fa fa-check' />} {labelForSortMode[mode]}{' '}
+                </React.Fragment>
+            ),
+            action: () => {
+                services.viewPreferences.updatePreferences({appDetails: {...prefs.appDetails, podView: {...podPrefs, sortMode: mode}}});
+                this.loader.reload();
+            }
+        }));
+    }
+    public render() {
+        return (
+            <DataLoader load={() => services.viewPreferences.getPreferences()}>
+                {prefs => {
+                    const podPrefs = prefs.appDetails.podView || ({} as PodViewPreferences);
+                    return (
+                        <React.Fragment>
+                            <div className='pod-view__settings'>
+                                <div className='pod-view__settings__section'>
+                                    GROUP BY:&nbsp;
+                                    <DropDownMenu
+                                        anchor={() => (
+                                            <button className='argo-button argo-button--base-o'>
+                                                {labelForSortMode[podPrefs.sortMode]}&nbsp;&nbsp;
+                                                <i className='fa fa-chevron-circle-down' />
+                                            </button>
+                                        )}
+                                        items={this.menuItemsFor(['node', 'parentResource', 'topLevelResource'], prefs)}
+                                    />
+                                </div>
+                            </div>
+                            <DataLoader
+                                ref={loader => (this.loader = loader)}
+                                load={async () => {
+                                    return podPrefs.sortMode === 'node' ? services.applications.getNodes(this.props.app.metadata.name) : null;
+                                }}>
+                                {nodes => {
+                                    return (
+                                        <div className='nodes-container'>
+                                            {this.processTree(podPrefs.sortMode, nodes).map(group => (
+                                                <div className='node white-box' key={group.name}>
+                                                    <div className='node__container--header' onClick={() => this.props.onItemClick(group.fullName)} style={{cursor: 'pointer'}}>
+                                                        <div style={{display: 'flex', alignItems: 'center'}}>
```

Comment:
Please let's make sure node containers have the same width:


![image](https://user-images.githubusercontent.com/426437/102123464-b94f4e00-3dfb-11eb-8d98-43ddefdb173b.png)